src/com/hms/flexyosisoftconnector/OsisoftJsonPayload.java

```diff
@@ -0,0 +1,382 @@
+          "Unable to initialize tags list."
+              + " It is possible no tags have been added to this device.");
+      tagNamesEncountered = new boolean[0];
+    }
+  }
+
+  /**
+   * Set the value of the tagId's index to true if it is being 'added' to this array
+   *
+   * @param tagId Id of the tag
+   */
+  private void addToTagsEncounteredArray(int tagId) {
+    int index = tagId - tagIndexOffset;
+    tagNamesEncountered[index] = true;
+  }
+
+  /**
+   * Return if the specified tag ID has been encountered
+   *
+   * @param tagId tag ID
+   * @return true if tag ID has been encountered
+   */
+  private boolean checkIfTagIdEncountered(int tagId) {
+    int index = tagId - tagIndexOffset;
+    return tagNamesEncountered[index];
+  }
+
+  /**
+   * Get the payload as a string
+   *
+   * @return payload string
+   */
+  public String getPayload() {
+    return payload.toString();
+  }
+
+  /**
+   * This function starts a new payload with the objects payload settings.
+   *
+   * <p>The status of the payload will be updated to show that the header information has been
+   * added.
+   */
+  public void startPayload() {
+
+    payload.clearString();
+    tagNamesEncountered = new boolean[tagsEncounteredArraySize];
+    tagPayloadArr = new TagPayload[tagsEncounteredArraySize];
+    payloadStartTimestamp = 0;
+    dataPointsEncounteredCount = 0;
+    String startOfPayload;
+
+    switch (communicationType) {
+      case OsisoftConfig.omf:
+        startOfPayload = PayloadBuilder.startOMFDataMessage();
+        break;
+      case OsisoftConfig.piwebapi:
+        startOfPayload = PayloadBuilder.startBatchOldFormat();
+        break;
+      default:
+        Logger.LOG_SERIOUS(comsErrMsg);
+        startOfPayload = "no communication type";
+        break;
+    }
+
+    boolean didAppendToPayload = payload.append(startOfPayload);
+    if (didAppendToPayload) {
+      status = PAYLOAD_HEADER_COMPLETE;
+    } else {
+      Logger.LOG_CRITICAL("An error occurred while appending the start of a payload.");
+    }
+  }
+
+  /**
+   * Append a data point to the payload
+   *
+   * @return true if datapoint added
+   */
+  public void appendDataPointToPayLoad(DataPoint dataPoint) {
+
+    addDataPoint(dataPoint);
+
+    didReachMaxDataPoints();
+
+    didTakeTooLong(dataPoint);
+
+    completePayloadAttempt();
+  }
+
+  private void didTakeTooLong(DataPoint data) {
+    long time = Long.parseLong(data.getTimeStamp());
+
+    /*
+     * The initial value of the start time is 0.
+     * set the value to 0 whenever the payload is complete.
+     *
+     * In either case, start a new time period and set the time.
+     */
+    if (payloadStartTimestamp == 0) {
+      payloadStartTimestamp = time;
+
+    } else if (payloadStartTimestamp < (time - PAY_LOAD_TIME_PERIOD_SECONDS)) {
+      // start a new payload due to reaching time period limit
+      status = PAYLOAD_VALUES_COMPLETE;
+    }
+  }
+
+  /**
+   * Adds a datapoint to the payload
+   */
```

Comment:
missing javadoc @ param (no space -- didnt want to tag the person named "param")